Source/engine.h

```diff
@@ -305,83 +312,134 @@ struct CelOutputBuffer {
 	}
 };
 
+/**
+ * Stores a CEL or CL2 sprite and its width(s).
+ *
+ * The data may be unowned.
+ * Eventually we'd like to remove the unowned version.
+ */
+class CelSprite {
+public:
+	CelSprite(std::unique_ptr<BYTE[]> data, int width)
+	    : data_(std::move(data))
+	    , data_ptr_(data_.get())
+	    , width_(width)
+	{
+	}
+
+	CelSprite(std::unique_ptr<BYTE[]> data, const int *widths)
+	    : data_(std::move(data))
+	    , data_ptr_(data_.get())
+	    , widths_(widths)
+	{
+	}
+
+	/**
+	 * Constructs an unowned sprite.
+	 * Ideally we'd like to remove all uses of this constructor.
+	 */
+	CelSprite(const BYTE *data, int width)
+	    : data_ptr_(data)
+	    , width_(width)
+	{
+	}
+
+	CelSprite(CelSprite &&) noexcept = default;
+	CelSprite &operator=(CelSprite &&) noexcept = default;
+
+	[[nodiscard]] const BYTE *Data() const
+	{
+		return data_ptr_;
+	}
+
+	[[nodiscard]] int Width(std::size_t frame = 1) const
+	{
+		return widths_ == nullptr ? width_ : widths_[frame];

```

Comment:
All frames have the same width, so i don't see the reason for this?

Reply:
That's not true, see e.g. InvItemWidth

Reply:
:(

i guess its only true for cl2?

Reply:
I'm not sure